extensions/github1s/src/adapters/ossinsight/data-source.ts

```diff
@@ -0,0 +1,180 @@
+/**
+ * @file trending page data source
+ * @author netcon
+ */
+
+import { joinPath } from '@/helpers/util';
+import { getCollections } from './interfaces';
+import { RankingLanguages, RankingPeriod } from './constants';
+import { DataSource, Directory, File, FileType, Promisable } from '../types';
+import {
+	createCollectionPageMarkdown,
+	createCollectionsListPageMarkdown,
+	createLanguageListPageMarkdown,
+	createRankingPageMarkdown,
+} from './templates';
+
+type StructureItem =
+	| {
+			type: FileType.Directory;
+			name: string;
+			children: StructureItem[] | (() => Promisable<StructureItem[]>);
+	  }
+	| {
+			type: FileType.File;
+			name: string;
+			content?: string | (() => Promisable<string>);
+	  };
+
+export class OSSInsightDataSource extends DataSource {
+	private static instance: OSSInsightDataSource | null = null;
+	private textEncodder = new TextEncoder();
+	private rootStructure: StructureItem = {
+		type: FileType.Directory,
+		name: '',
+		children: [
+			{
+				type: FileType.Directory,
+				name: '.vscode',
+				children: [
+					{
+						type: FileType.File,
+						name: 'settings.json',
+						content: () =>
+							JSON.stringify({
+								'files.exclude': { '.vscode': true },
+								'markdown.preview.doubleClickToSwitchToEditor': false,
+								'workbench.editorAssociations': {
+									'*.md': 'vscode.markdown.preview.editor',
+								},
+							}),
+					},
+				],
+			},
+			{
+				type: FileType.Directory,
+				name: 'collections',
+				children: () =>
+					getCollections().then((collections) => {
+						const collectionsSet = new Set(collections.map((collection) => collection.name));
+						return Array.from(collectionsSet).map((collection) => ({
+							type: FileType.File,
+							name: `${collection}.md`,
+							content: () => createCollectionPageMarkdown(collection),
+						}));
+					}),
+			},
+			{
+				type: FileType.Directory,
+				name: 'languages',
+				children: () =>
+					RankingLanguages.map((language) => {
+						return {
+							type: FileType.Directory,
+							name: language,
+							children: [
+								{
+									type: FileType.File,
+									name: `${language}_Today.md`,
+									content: () => createRankingPageMarkdown(RankingPeriod.Today, language),
+								},
+								{
+									type: FileType.File,
+									name: `${language}_ThisWeek.md`,
+									content: () => createRankingPageMarkdown(RankingPeriod.ThisWeek, language),
+								},
+								{
+									type: FileType.File,
+									name: `${language}_ThisMonth.md`,
+									content: () => createRankingPageMarkdown(RankingPeriod.ThisMonth, language),
+								},
+							],
+						};
+					}),
+			},
+			{
+				type: FileType.File,
+				name: 'Collections.md',
+				content: () => createCollectionsListPageMarkdown(),
+			},
+			{
+				type: FileType.File,
+				name: 'Languages.md',
+				content: () => createLanguageListPageMarkdown(),
+			},
+			{
+				type: FileType.File,
+				name: 'README.md',
```

Comment:
We can name it `TrendingToday.md` but, we need to preview this file also. Then it will require a change to that functionality. Let's keep it README, comment is just for future references.